pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/BaseJdbcAutoSchemaSink.java

```diff
@@ -185,5 +187,35 @@ private void fillKeyValueSchemaData(org.apache.pulsar.client.api.Schema<GenericO
                         + " with KeyValueSchema");
         }
     }
+
+    private static Object convertAvroField(Object avroValue, Schema schema) {
+        switch (schema.getType()) {
+            case NULL:
+            case INT:
+            case LONG:
+            case DOUBLE:
+            case FLOAT:
+            case BOOLEAN:
+                return avroValue;
+            case BYTES:
+                // system default charset
+                return new String((byte[]) avroValue);
```

Comment:
This looks wrong. In jdbc we have the array type. We should use that

Reply:
From [Avro doc](https://dmg.org/pfa/docs/avro_types/index.html)

>The bytes type is referred to as "bytes" and it can accept any byte sequence. It is therefore a generalization of a string (though neither is a subtype of the other).
Values of type "bytes" are not arrays and cannot be changed in-place (all PFA values are immutable).

Reply:
just realized that you meant bytes array type, like bytea in Postgre, will fix it